README.md

```diff
@@ -172,6 +172,45 @@ TID is a 32-bit integer to identify threads created with `wasi_thread_spawn`.
     For example, it can be used to indicate the main thread, which doesn't
     have a TID in the current version of this proposal.
 
+### Process
+
+* A process is a group of threads.
+
+* The main thread starts with a process which only contains
+  the main thread.
+
+* Threads created by a thread in a process using `wasi_thread_spawn`
+  is added to the process.
+
+* When a thread is terminated, it's removed from the process.
+
+### Voluntary thread termination
+
+A thread can terminate itself voluntarily by returning from
+`wasi_thread_start`.
+
+### Changes to WASI `proc_exit`
+
+With this proposal, the `proc_exit` function takes extra responsibility
+to terminate all threads in the process, not only the calling one.
+
+Any of threads in the process can call `proc_exit`.
+
+### Traps
+
+When a thread caused a trap, it terminates all threads in the process
+similarly to `proc_exit`.
+
+### Process exit status
+
+If one or more threads call WASI `proc_exit` or raise a trap,
+one of them is chosen by the runtime to represent the exit status
+of the process.
+It's non deterministic which one is chosen.
+
+If the process gets empty without involving `proc_exit` or a trap,
```

Comment:
"gets empty" sounds a little odd to me.  How about "If the last remaining thread in the process returns from its entry point without trapping or calling `proc_exit` it is treated as if `proc_exit` was called with exit code 0".

Reply:
it isn't only about the last thread.
eg. if the second last thread terminated itself with a trap and the last thread returned from `wasi_thread_start`, i still want to make the trap represent the process exit status.

also, i want to avoid mentioning the entry point in this sentence because it can involve another topic: https://github.com/WebAssembly/wasi-threads/issues/21

how about:
```
If all the threads in the process have been terminated without calling 
`proc_exit` or raising a trap, it's treated as if the last thread called 
`proc_exit` with exit code 0.
```

Reply:
Sure thats sounds fine.   Perhaps we can define "normal termination" and "early termination" and say "If all the threads in the process terminate normally"?

Reply:
> Sure thats sounds fine.

done

> Perhaps we can define "normal termination" and "early termination" and say "If all the threads in the process terminate normally"?

i feel it confusing as calling exit() is quite normal.

Reply:
I think its reasonable to refer to "exit()" as early and/or abnormal termination.